src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -185,17 +188,25 @@ public async Task<JObject> Resolve(string varName, CancellationToken token)
             if (scopeCache.ObjectFields.TryGetValue(varName, out JObject valueRet))
                 return await GetValueFromObject(valueRet, token);
 
-            string[] parts = varName.Split(".");
+            string[] parts = varName.Split(".", StringSplitOptions.TrimEntries);
             if (parts.Length == 0)
                 return null;
 
             JObject retObject = await ResolveAsLocalOrThisMember(parts[0]);
+            bool isRootNullSafe = parts[0].LastOrDefault() == '?';
```

Comment:
After changing it turns out we have some failing tests. When parts[0] is empty and it can happen when we are testing with `expression = ".a."` then we end with parts = { "", "a", "" }. So indexing 0th element is not safe. When we use FirstOrDefault we are sure not to index wrong. The expressions cannot be too long as they are manually provided by users so iterating through one part to get the last element does not seem painful here because we would have to check for nullOrEmpty in the indexing scenario additionally. What do you think?

Reply:
Shouldn't we fail in that case? `.a.` isn't a valid expression by itself.

Reply:
Actually yes, it's a good idea to finish the processing early with throwing "RefError". I decided to do the same for `parts.Count == 0` because it's a similar situation and the expected result does not change.